includes/data-port/import-tasks/class-sensei-import-file-process-task.php

```diff
@@ -78,22 +92,57 @@ public function run() {
 			return;
 		}
 
-		$lines = $this->reader->read_lines();
+		if ( ! $this->reader->is_completed() ) {
+			$lines = $this->reader->read_lines();
+
+			$current_line = $this->completed_lines;
+
+			foreach ( $lines as $line_data ) {
+				$current_line++;
 
-		$current_line = $this->completed_lines;
+				$this->process_line( $current_line + 1, $line_data );
+			}
 
-		foreach ( $lines as $line ) {
-			$this->process_line( ++$current_line, $line );
+			$this->completed_lines = $this->reader->get_completed_lines();
+			$this->total_lines     = $this->reader->get_total_lines();
+		} elseif ( $this->reader->is_completed() ) {
+			// Running this in an else so that post process tasks run in a fresh batch.
+			$this->run_post_process_tasks();
```

Comment:
There are multiple early returns both in `Sensei_Data_Port_Job::run()` and `Sensei_Import_File_Process_Task::run()` on `$task->is_completed()`, which is set to `$this->reader->is_completed()` - the post process tasks never run.

Reply:
Ahh, missed it in the constructor. 9a21c53b5d6e44750febf3e43314230e6d0fcdba

Alternatively, we could just get rid of the `is_completed` var and put the check in the method.